From d65007581584ed54d8805e936d500b1cb9e3d37e Mon Sep 17 00:00:00 2001
From: Zhang Xiao <xiao.zhang@windriver.com>
Date: Mon, 1 Jun 2020 00:30:48 -0700
Subject: [PATCH] Revert "net: phy: Avoid multiple suspends"

This reverts commit 5d767568e147bfce62749d39d3e9e9936cb0ab0d.

Commit 5d767568 based on upstream commit 503ba7c69 that fixed
an issue caused by upstream commit 803dd9c77ac3. While both
nxp-imx6 and  nxp-imx8 haven't integrated upstream commit
803dd9c77. So revert it.

Signed-off-by: Zhang Xiao <xiao.zhang@windriver.com>
---
 drivers/net/phy/phy_device.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/net/phy/phy_device.c b/drivers/net/phy/phy_device.c
index a18500f5b844..e18f8eb593d1 100644
--- a/drivers/net/phy/phy_device.c
+++ b/drivers/net/phy/phy_device.c
@@ -276,8 +276,7 @@ static bool mdio_bus_phy_may_suspend(struct phy_device *phydev)
 	if (device_may_wakeup(&netdev->dev))
 		return false;
 
-out:
-	return !phydev->suspended;
+	return true;
 }
 
 static int mdio_bus_phy_suspend(struct device *dev)
-- 
2.17.1

